src/main/scala/chisel3/ltl/LTL.scala

```diff
@@ -363,6 +363,29 @@ sealed abstract class AssertPropertyLike {
     verif.property := clocked.inner
   }
 
+  /**
+    * Overloads the apply method to allow for Bool expressions to be used in AssertProperty.
+    * This avoids having to import Sequence on the user side to get the implicit conversion.
+    */
```

Comment:
I don't think there's really a reason to mention overloading in the ScalaDoc, from the user's perspective this is just the API they should call (I would just basically copy-paste the ScalaDoc on the main apply function although it would be good to turn those bullets into actual Scaladoc (using `@param`).

We also need a form that takes `clock` and `disable`. This is annoying since we can't have another apply with default arguments, but maybe just have 3, one taking the `prop`, one taking the `prop` and `label`, and one taking all of the arguments.